cmake/setup_compiler_flags.cmake

```diff
@@ -68,6 +68,7 @@
 #
 
 foreach(build ${DEAL_II_BUILD_TYPES})
+  set(CMAKE_TRY_COMPILE_CONFIGURATION ${build})
```

Comment:
Should we unset the variable again after the `check_compiler_setup` call? Just to make sure that it doesn't influence any other configuration check.